src/protocol/ipa/mod.rs

```diff
@@ -155,24 +156,60 @@ impl<F: Field + Sized, T: Arithmetic<F>> Resharable<F> for IPAModulusConvertedIn
     }
 }
 
+/// Describes a request to execute IPA protocol. Consists of inputs for the computation
+/// and query parameters
+pub struct Query<'a, F: Field, MK: BitArray = MatchKey, BK: BitArray = BreakdownKey> {
+    input: &'a [IPAInputRow<F, MK, BK>],
```

Comment:
I think that you should remove this (keep it separate) and then name the struct IpaConfig or something.